src/usbsettings.h

```diff
@@ -40,29 +41,36 @@ class USBSettings: public QObject
 
     Q_ENUMS(Mode)
 
-    Q_PROPERTY(Mode currentMode READ currentMode NOTIFY currentModeChanged)
+    Q_PROPERTY(Mode currentMode READ currentMode WRITE setCurrentMode NOTIFY currentModeChanged)
     Q_PROPERTY(Mode defaultMode READ defaultMode WRITE setDefaultMode NOTIFY defaultModeChanged)
     Q_PROPERTY(QList<int> supportedUSBModes READ supportedUSBModes NOTIFY supportedUSBModesChanged)
 
 public:
+    /**
+     * Keep this in sync with usb_moded-dbus.h (for states) and usb_moded-modes.h (for modes),
+     * existing enum values taken from legacy qmsystem2 enum mapping for compatibility
+     **/
     enum Mode {
-        Connected = MeeGo::QmUSBMode::Connected,
-        DataInUse = MeeGo::QmUSBMode::DataInUse,
-        Disconnected = MeeGo::QmUSBMode::Disconnected,
-        MassStorage = MeeGo::QmUSBMode::MassStorage,
-        ChargingOnly = MeeGo::QmUSBMode::ChargingOnly,
-        OviSuite = MeeGo::QmUSBMode::OviSuite,
-        ModeRequest = MeeGo::QmUSBMode::ModeRequest,
-        Ask = MeeGo::QmUSBMode::Ask,
-        Undefined = MeeGo::QmUSBMode::Undefined,
-        SDK = MeeGo::QmUSBMode::SDK,
-        Developer = MeeGo::QmUSBMode::Developer,
-        MTP = MeeGo::QmUSBMode::MTP,
-        Adb = MeeGo::QmUSBMode::Adb,
-        Diag = MeeGo::QmUSBMode::Diag,
-        Host = MeeGo::QmUSBMode::Host,
-        Charger = MeeGo::QmUSBMode::Charger,
-        ConnectionSharing = MeeGo::QmUSBMode::ConnectionSharing
+        // States (from usb_moded-dbus.h)
+        Connected = 0,
+        DataInUse = 1,
+        Disconnected = 2,
+        ModeRequest = 6,
+
+        // Modes (from usb_moded-modes.h)
+        MassStorage = 3,
+        ChargingOnly = 4,
+        Ask = 7,
+        Undefined = 8,
```

Comment:
Undefined happens when nothing is connected and no state is set, or when an error happens during the mode setting. Of course there are signals to indicate the changes/error, but the state will be undefined to point out that the actual state is unknown.